### PR TITLE
scripts/genproto: use 'gopath.proto' for $GOPATH

### PR DESCRIPTION
**Pull request by gyuho**
Fix https://github.com/coreos/etcd/issues/6552.

This changes `genproto.sh` script to just use local GOPATH.
Regenerated protos have different bytes because of `gofmt` changes.
For next runs, there shouldn't be difference.

If any, I will file an issue to `gogo/protobuf`.

/cc @heyitsanthony @xiang90

### CODE REVIEW

**Comment by heyitsanthony** [comment]
This is kind of godep-ish in the way it trashes the GOPATH (running it failed for me since I had an old version of the grpc gateway). I tried it with a fresh gopath (no contents), but it complains about not finding etcd so this would presumably break on out-of-path builds.

Maybe this can create `GOPATH=gopath.proto` instead of `gopath` and delete it when it's done so there's no conflict with `./build`?

**Comment by gyuho** [comment]
@heyitsanthony You are right. This should not depend on the local machine's GOPATH.

Fixed as you suggested.

Thanks.

File: scripts/genproto.sh [hunk 1]
```diff
@@ -76,6 +76,8 @@ protoc -I. \
     --swagger_out=logtostderr=true:./Documentation/dev-guide/apispec/swagger/. \
     ./etcdserver/etcdserverpb/rpc.proto
 
+rm -rf ${GOPATH}
```
Comment (heyitsanthony) [inline scripts/genproto.sh]: just in case something meddles with the gopath in between, do `rm -rf ${PWD}/gopath.proto` here?

**Comment by gyuho** [comment]
@heyitsanthony Just fixed. Tried this script and output looks good.

PTAL. Thanks!

**Comment by heyitsanthony** [comment]
lgtm

**Pull request merged**